tests/TestCase/Queue/ProcessorTest.php

```diff
@@ -129,6 +130,43 @@ public function testProcessNotValidCallable()
         $this->assertSame('Processor.message.invalid', $events[1]->getName());
     }
 
+    /**
+     * When processMessage() throws an exception, test that
+     * requeue will return.
+     *
+     * @return void
+     */
+    public function testProcessWillRequeueOnException()
+    {
+        $method = 'processReturnNull';
```

Comment:
Couldn't you also have a method on the test that throws an exception which removes the need to have a mock on line 157?

Reply:
Yes I could.